coalib/io/Directory.py

```diff
@@ -0,0 +1,28 @@
+import os
+
+from coalib.parsing.Globbing import relative_recursive_glob
+from coala_utils.decorators import generate_eq
+
+
+@generate_eq('path')
+class Directory:
+    def __init__(self, path):
+        self._path = os.path.abspath(path)
+        self._parent = os.path.abspath(os.path.dirname(self._path))
+
+    def children(self):
+        elements = relative_recursive_glob(self._path, '**')
```

Comment:
can't we use `os.listdir`? I assume that's way faster because it's not evaluating globs^^

Reply:
`os.listdir` only gives us the children contained in the parent directories and does not go down into the sub directories.
Since the directory caching strategy is not clear I'm not sure whether to get all the children recursively or just the ones one level down the parent.

What do you think? Should I still change it to `listdir` or keep the current code as it is or have two different methods one recursive and one non-recursive?

Reply:
Ahh you recurse. Definitely make it top-level only right now, always recursing into subdirs is a performance penalty. I'm not sure how much our globbing is caching directory contents, but I wouldn't rely on that^^